Source/Request.swift

```diff
@@ -469,6 +469,21 @@ public class Request {
         fatalError("Subclasses must override.")
     }
 
+    // MARK: Request State
+
+    func attemptToTransitionTo(_ state: Request.State) -> Bool {
+        var transitioned = false
+
+        protectedMutableState.write { mutableState in
+            guard mutableState.state.canTransitionTo(state) else { return }
+
+            mutableState.state = state
```

Comment:
We should remove the ability to modify `state` this way in a future PR. Instead, `canTransitionTo` should be a `transitionTo` API that only allows transitioning to a valid state.